### PR TITLE
v1alpha1 validators: catch possible reqState nil case

### LINKED ISSUE #9161: Opening prysm ui while processing deposits
# 🐞 Bug Report

### Description

When opening prysm UI while beacon is still processing deposit logs and error appears in the logs.

### Has this worked before in a previous version?

don't know

## 🔬 Minimal Reproduction

Start beacon from scratch, no db. Prysm will start processing deposits. Then open up Prysm UI

## 🔥 Error

<pre><code>
time="2021-07-07 16:48:00" level=error msg="gRPC panicked!" error="runtime error: invalid memory address or nil pointer dereference" stack="goroutine 80441 [running]:
runtime/debug.Stack(0xc0000b0000, 0x5d857e, 0x5)
GOROOT/src/runtime/debug/stack.go:24 +0x9f
github.com/prysmaticlabs/prysm/shared/traceutil.RecoveryHandlerFunc(0x952498, 0xc01d9ab740, 0x3ae9a0, 0x240e4d0, 0x0, 0x0)
shared/traceutil/recovery_interceptor_option.go:29 +0x154
github.com/grpc-ecosystem/go-grpc-middleware/recovery.recoverFrom(0x952498, 0xc01d9ab740, 0x3ae9a0, 0x240e4d0, 0x7a6ce0, 0x1d10600, 0xc00c983280)
external/com_github_grpc_ecosystem_go_grpc_middleware/recovery/interceptors.go:61 +0x5a
github.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1.1(0xc00c983a2e, 0x952498, 0xc01d9ab740, 0xc000116920, 0xc00c983ab0)
external/com_github_grpc_ecosystem_go_grpc_middleware/recovery/interceptors.go:29 +0x71
panic(0x3ae9a0, 0x240e4d0)
GOROOT/src/runtime/panic.go:965 +0x1b9
github.com/prysmaticlabs/prysm/beacon-chain/rpc/beacon.(*Server).ListValidators(0xc00013e500, 0x952498, 0xc01d9abda0, 0xc01d9b2e80, 0xc00013e500, 0xc000f60ed0, 0x7a94e0)
beacon-chain/rpc/beacon/validators.go:236 +0x3b0
github.com/prysmaticlabs/prysm/proto/eth/v1alpha1._BeaconChain_ListValidators_Handler.func1(0x952498, 0xc01d9abda0, 0x515ca0, 0xc01d9b2e80, 0x952498, 0xc01d9ab740, 0x3abe20, 0x2cd2028)
bazel-out/k8-opt/bin/proto/eth/v1alpha1/go_proto_/github.com/prysmaticlabs/prysm/proto/eth/v1alpha1/beacon_chain.pb.go:5470 +0x8b
github.com/prysmaticlabs/prysm/beacon-chain/rpc.(*Service).validatorUnaryConnectionInterceptor(0xc000f60e70, 0x952498, 0xc01d9abda0, 0x515ca0, 0xc01d9b2e80, 0xc010a8aae0, 0xc01d9a7038, 0x2cd1a00, 0x952498, 0xc01d9abda0, ...)
beacon-chain/rpc/service.go:340 +0x79
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x952498, 0xc01d9abda0, 0x515ca0, 0xc01d9b2e80, 0x5e35f9, 0xd, 0x624a33, 0x31)
external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0xc010a8aae0, 0xc010a8ab00, 0x3, 0x3, 0x95cd58, 0xc01218ca80)
external/com_github_grpc_ecosystem_go_grpc_middleware/tracing/opentracing/server_interceptors.go:34 +0x108
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0x31, 0xc0128dd7c0, 0x12c94b0, 0xc0153070b0)
external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).UnaryServerInterceptor.func1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0xc010a8aae0, 0xc010a8ab20, 0x2264c9d, 0xc015302750, 0x325460, 0xc01d9b0a40)
external/com_github_grpc_ecosystem_go_grpc_prometheus/server_metrics.go:107 +0xb0
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0x497d60, 0xa2, 0x3f2faadaace, 0x7ff31402c718)
external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0xc010a8aae0, 0xc010a8ab40, 0x0, 0x0, 0x0, 0x0)
external/com_github_grpc_ecosystem_go_grpc_middleware/recovery/interceptors.go:33 +0xcb
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0xc000101400, 0x0, 0xc00c983b30, 0x12d4778)
external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:25 +0x63
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1(0x952498, 0xc01d9ab740, 0x515ca0, 0xc01d9b2e80, 0xc010a8aae0, 0xc01d9a7038, 0xc00bf79ba0, 0x13d9c06, 0x4d43e0, 0xc01d9ab740)
external/com_github_grpc_ecosystem_go_grpc_middleware/chain.go:34 +0xd7
github.com/prysmaticlabs/prysm/proto/eth/v1alpha1._BeaconChain_ListValidators_Handler(0x5a2620, 0xc00013e500, 0x952498, 0xc01d9ab740, 0xc01218c300, 0xc00c2fe510, 0x952498, 0xc01d9ab740, 0xc0001e4140, 0x9d)
bazel-out/k8-opt/bin/proto/eth/v1alpha1/go_proto_/github.com/prysmaticlabs/prysm/proto/eth/v1alpha1/beacon_chain.pb.go:5472 +0x150
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000b9ca80, 0x9639b8, 0xc0002ca900, 0xc0015539e0, 0xc00c2fea50, 0x24286e0, 0x0, 0x0, 0x0)
external/org_golang_google_grpc/server.go:1217 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000b9ca80, 0x9639b8, 0xc0002ca900, 0xc0015539e0, 0x0)
external/org_golang_google_grpc/server.go:1540 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc006eb4060, 0xc000b9ca80, 0x9639b8, 0xc0002ca900, 0xc0015539e0)
external/org_golang_google_grpc/server.go:878 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
external/org_golang_google_grpc/server.go:876 +0x1fd
"</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
Ubuntu 20.4
  </code>
</pre>

**What version of Prysm are you running? (Which release)**

<pre>
  <code>
1.3.11
  </code>
</pre>

with [Stereum Node Setup](https://stereum.net/ethereum-node-setup/)

### PR DESCRIPTION
**Pull request by KaanKC**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**

The `reqState` variable can be `nil` at certain times even if the underlying request didn't return an error which causes a runtime error. This PR fixes the issue by verifying that the value isn't `nil` 

**Which issues(s) does this PR fix?**

Fixes #9161  

**Other notes for review**

None.

### CODE REVIEW

File: beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go [hunk 1]
```diff
@@ -228,6 +228,9 @@ func (bs *Server) ListValidators(
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not get requested state: %v", err)
 	}
+	if reqState == nil {
```
Comment (nisdas) [inline beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go:231]:
```suggestion
	if reqState == nil || reqState.isNil() {
```

Comment (KaanKC) [reply beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go]: right, thanks

File: beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go [hunk 2]
```diff
@@ -228,6 +228,9 @@ func (bs *Server) ListValidators(
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Could not get requested state: %v", err)
 	}
+	if reqState == nil || reqState.IsNil() {
+		return nil, status.Error(codes.Internal, "requested state is nil")
```
Comment (terencechain) [inline beacon-chain/rpc/prysm/v1alpha1/beacon/validators.go:232]:
```suggestion
		return nil, status.Error(codes.Internal, "Requested state is nil")
```

File: beacon-chain/rpc/prysm/v1alpha1/beacon/validators_test.go [hunk 3]
```diff
@@ -433,6 +433,40 @@ func TestServer_ListValidators_CannotRequestFutureEpoch(t *testing.T) {
...
+		},
+		StateGen: &stategen.MockStateManager{
+			StatesBySlot: map[types.Slot]state.BeaconState{
+				0: nil,
+			},
+		},
+	}
+	// request uses HeadFetcher to get reqState.
+	req1 := &ethpb.ListValidatorsRequest{PageToken: strconv.Itoa(1), PageSize: 100}
+	wanted := "requested state is nil"
```
Comment (terencechain) [inline beacon-chain/rpc/prysm/v1alpha1/beacon/validators_test.go:456]:
```suggestion
	wanted := "Requested state is nil"
```

**Review by terencechain — APPROVED** [review]

**Comment by KaanKC** [comment]
Buildkite failed although it ran through prior to @terencechain 's commit. I don't see why tho since he only capitalized the first word of the error message.

**Comment by rkapka** [comment]
> Buildkite failed although it ran through prior to @terencechain 's commit. I don't see why tho since he only capitalized the first word of the error message.

Unfortunately our end-to-end CI step is flaky. It's almost certain that your code is OK. I restarted the test.

**Comment by KaanKC** [comment]
Thanks :)

**Pull request merged**